addon-test-support/test-link.ts

```diff
@@ -7,11 +7,23 @@ import { Link } from 'ember-link';
 
 export default class TestLink extends Link {
   // Overwritable properties
+  // @typescript-eslint/ban-ts-comment
+  // @ts-expect-error this fails consumer packages
   @tracked isActive = false;
+  // @typescript-eslint/ban-ts-comment
+  // @ts-expect-error this fails consumer packages
   @tracked isActiveWithoutQueryParams = false;
+  // @typescript-eslint/ban-ts-comment
+  // @ts-expect-error this fails consumer packages
   @tracked isActiveWithoutModels = false;
+  // @typescript-eslint/ban-ts-comment
+  // @ts-expect-error this fails consumer packages
   @tracked isEntering = false;
+  // @typescript-eslint/ban-ts-comment
+  // @ts-expect-error this fails consumer packages
   @tracked isExiting = false;
+  // @typescript-eslint/ban-ts-comment
+  // @ts-expect-error this fails consumer packages
```

Comment:
I suspect that this will not actually work.

This addon has a build step that is executed before publishing to npm:

https://github.com/buschtoens/ember-link/blob/c8497dca8c5ffc559987a5bf02e4a465ed500e0d/package.json#L28-L29

This generates the respective `.js` and `.d.ts` files.

`tsc` does not include `// @ts-*` comments in the resulting `.d.ts` files:

```ts
import Transition from '@ember/routing/-private/transition';
import { Link } from 'ember-link';
export default class TestLink extends Link {
    isActive: boolean;
    isActiveWithoutQueryParams: boolean;
    isActiveWithoutModels: boolean;
    isEntering: boolean;
    isExiting: boolean;
    url: string;
    onTransitionTo?(): void;
    onReplaceWith?(): void;
    get qualifiedRouteName(): string;
    transitionTo(event?: Event): Transition;
    replaceWith(event?: Event): Transition;
    private _preventTransitionOut;
    private _createDummyTransition;
}
```

So those comments actually have no effect for users.

(FWIW this worked as a `patch-package` patch in our local repo, because we are depending on an older version that doesn't yet pre-compile.)

Reply:
I'll try to think of a backwards-compatible fix.

Reply:
Ah good point, I forgot about extraction steps in transpilation.